libsolidity/interface/FileReader.cpp

```diff
@@ -31,9 +33,20 @@ using std::string;
 namespace solidity::frontend
 {
 
+void FileReader::setBasePath(boost::filesystem::path const& _path)
+{
+	if (_path.empty())
+		m_basePath = "";
+	else
+		m_basePath = normalizeCLIPathForVFS(_path);
+}
+
 void FileReader::setSource(boost::filesystem::path const& _path, SourceCode _source)
 {
-	m_sourceCodes[_path.generic_string()] = std::move(_source);
+	boost::filesystem::path normalizedPath = normalizeCLIPathForVFS(_path);
+	boost::filesystem::path prefix = (m_basePath.empty() ? normalizeCLIPathForVFS(".") : m_basePath);
```

Comment:
I like it better with parentheses and I always use them when assigning a ternary expression. I find this more redaable because they group things visually. Without them the left-hand-side blends a bit too much with the rest of the expression.